src/client/scripts/game/misc/onlinegame.js

```diff
@@ -370,6 +373,47 @@ const onlinegame = (function(){
         stopOpponentAFKCountdown(); // The opponent is no longer AFK if they were
         flashTabNameYOUR_MOVE(true);
         scheduleMoveSound_timeoutID();
+
+        handlePremove();
+    }
+    /**
+     * Submits the next premove to the server if it is legal.
+     * Otherwise deletes the queue.
+     */
+    function handlePremove() {
```

Comment:
This is weird to put this function in onlinegame.js. Let's create a new script called `premove.js` and put all the premove logic in there to keep it separated from the rest of the code!